packages/replay/src/main.ts

```diff
@@ -245,35 +248,64 @@ async function doUploadRecording(
   const metadata = recording.metadata
     ? await client.buildRecordingMetadata(recording.metadata, { verbose })
     : null;
-  const { recordingId, uploadLink } = await client.connectionBeginRecordingUpload(
-    recording.id,
-    recording.buildId!,
-    size
-  );
-  debug(`Created remote recording ${recordingId}`);
-  if (metadata) {
-    try {
-      await client.setRecordingMetadata(recordingId, metadata);
-    } catch (e) {
-      handleUploadingError(`Failed to set recording metadata ${e}`, strict, verbose, e);
+
+  const setMetadata = async (recordingId: string, metadata: RecordingMetadata | null) => {
+    if (metadata) {
+      try {
+        await client.setRecordingMetadata(recordingId, metadata);
+      } catch (e) {
+        handleUploadingError(`Failed to set recording metadata ${e}`, strict, verbose, e);
+      }
     }
-  }
+  };
 
-  addRecordingEvent(dir, "uploadStarted", recording.id, {
-    server,
-    recordingId,
-  });
+  let recordingId: string;
+  if (size > MIN_MULTIPART_UPLOAD_SIZE && process.env.REPLAY_MULTIPART_UPLOAD) {
+    const requestPartChunkSize =
```

Comment:
These are complicated/long enough that it might warrant refactoring them into separate methods.

```
let recordingId = (size > MIN_MULTIPART_UPLOAD_SIZE && process.env.REPLAY_MULTIPART_UPLOAD) ?
  multiPartRecordingUpload(...) :
  recordingUpload();
```